lib/kits/core-ui/paged-publication.js

```diff
@@ -0,0 +1,206 @@
+            };
+
+            if (!options.id) {
+                options.id =
+                    await pagedPublicationObj.fetchLatestPublicationId();
+            }
+        },
+
+        start: () => {
+            const bootstrapper = new PagedPublicationKit.Bootstrapper(options);
+
+            bootstrapper.fetch((err, data) => {
+                if (!err) {
+                    sgnData = data;
+                    sgnViewer = bootstrapper.createViewer(data);
+
+                    const headerEl = options.el.querySelector('.sgn__header');
+                    headerEl?.classList.add('sgn-animate-header');
+
+                    sgnViewer.bind('hotspotClicked', (hotspot) => {
+                        const clientLocal = clientLocalStorage.get(
+                            'paged-publication-saved-offers'
+                        );
+                        const shopListOffer = {
+                            name: hotspot.heading,
+                            price: hotspot.offer?.pricing?.price,
+                            is_ticked: false
+                        };
+
+                        if (!clientLocal) {
+                            clientLocalStorage.set(
+                                'paged-publication-saved-offers',
+                                [shopListOffer]
+                            );
+                        } else {
+                            clientLocal.push(shopListOffer);
+                            clientLocalStorage.set(
+                                'paged-publication-saved-offers',
+                                clientLocal
+                            );
+                        }
+
+                        pagedPublicationObj.animateShoppingListCounter();
+                    });
+
+                    sgnViewer.start();
+                    // Fetch hotspots after rendering the viewer as they are not critical for initial render.
+                    bootstrapper.fetchHotspots((err2, hotspots) => {
+                        if (!err2) {
+                            bootstrapper.applyHotspots(sgnViewer, hotspots);
+                        }
+                    });
+                } else {
+                    console.error(err);
+                }
+            });
+        },
+
+        fetchLatestPublicationId: () => {
+            return new Promise((resolve) => {
+                request(
+                    {
+                        url: '/v2/catalogs',
+                        qs: {
+                            dealer_id: scriptEls.businessId,
+                            order_by: '-publication_date',
+                            limit: 1
+                        }
+                    },
+                    (err, publications) => {
+                        if (!err) {
+                            const config = {
+                                publicationId: publications[0].id
+                            };
+                            SGN.config.set(config);
```

Comment:
I think this is polluting the global config, plus I can't see where you're using this `publicationId` anywhere?